lib/bloc/csv_exporter.dart

```diff
@@ -0,0 +1,66 @@
+import 'dart:io';
+
+import 'package:breez/bloc/account/account_model.dart';
+import 'package:breez/utils/date.dart';
+import 'package:csv/csv.dart';
+import 'package:flutter/foundation.dart';
+import 'package:intl/intl.dart';
+import 'package:path_provider/path_provider.dart';
+
+class CsvExporter {
+  final List paymentList;
+  final PaymentFilterModel filter;
+
+  CsvExporter(this.paymentList, this.filter);
+
+  Future export() async {
+    return await _saveCsvFile(const ListToCsvConverter().convert(_generateList()));
+  }
+
+  List _generateList() {
+    List<List<dynamic>> paymentList = new List.generate(this.paymentList.length, (index) {
+      List paymentItem = new List();
+      PaymentInfo paymentInfo = this.paymentList.elementAt(index);
+      paymentItem
+          .add(DateUtils.formatYearMonthDayHourMinute(DateTime.fromMillisecondsSinceEpoch(paymentInfo.creationTimestamp.toInt() * 1000)));
+      paymentItem.add(paymentInfo.title);
+      paymentItem.add(paymentInfo.description);
+      paymentItem.add(paymentInfo.destination);
+      paymentItem.add(paymentInfo.amount.toString());
+      paymentItem.add(paymentInfo.preimage);
+      paymentItem.add(paymentInfo.paymentHash);
+      return paymentItem;
+    });
+    paymentList.insert(0, ["Date & Time", "Title", "Description", "Node ID", "Amount", "Preimage", "TX Hash"]);
+    return paymentList;
+  }
+
+  Future<String> _saveCsvFile(String csv) async {
+    String filePath = await _createCsvFilePath();
+    final file = File(filePath);
+    await file.writeAsString(csv);
+    return file.path;
+  }
+
+  Future<String> _createCsvFilePath() async {
+    final directory = await getApplicationDocumentsDirectory();
```

Comment:
Better to use temp directory.

Reply:
[Done](https://github.com/breez/breezmobile/pull/203/commits/015c455ec70c6f70ddce61f717970ce8d71b735d)